azure-iot-device/azure/iot/device/iothub/auth/iotedge_authentication_provider.py

```diff
@@ -112,7 +113,9 @@ def get_trust_bundle(self):
         :raises: IoTEdgeError if unable to retrieve the certificate.
         """
         r = requests.get(
-            self.workload_uri + "trust-bundle", params={"api-version": self.api_version}
+            self.workload_uri + "trust-bundle",
+            params={"api-version": self.api_version},
+            headers={"User-Agent": urllib.parse.quote_plus(constant.USER_AGENT)},
```

Comment:
>headers={"User-Agent": urllib.parse.quote_plus(constant.USER_AGENT)}, [](start = 12, length = 69)

Is there a reason we need this, or are you just following some "user-agent everywhere" mantra?  I think either is OK, but it may not be necessary to have this, fewer lines of code to break, yadda yadda yadda.

Reply:
Yes, in fact, this is specifically the entire reason for this PR. The lack of User-Agent was causing Edge to throw errors (!!!!)

---
In reply to: [307825867](https://github.com/Azure/azure-iot-sdk-python-preview/pull/147#discussion_r307825867) [](ancestors = 307825867)